### ISSUE TITLE
tr: Illegal byte sequence

### ISSUE THREAD

**Issue by riywo** [arun-gupta/kubernetes-aws-workshop#196]
https://github.com/arun-gupta/kubernetes-aws-workshop/tree/master/cluster-install

In my macOS Sierra bash terminal, the command below doesn't work:

```bash
$ export S3_BUCKET=example-state-store-$(cat /dev/random | LC_CTYPE=C tr -dc "[:alpha:]" | tr '[:upper:]' '[:lower:]' | head -c 32)
tr: Illegal byte sequence
```

Because of that, my `$S3_BUCKET` is not good.

```bash
$ echo $S3_BUCKET
example-state-store-
```

**Comment by arun-gupta** [comment]
@wombat can you take a look at this? can you talk to @riywo to clarify this?

The command works on my machine.

**Comment by riywo** [comment]
Worked with `LC_ALL=C` instead of `LC_CTYPE=C`. cf. https://coderwall.com/p/4zux3a/random-string-generator-bash

**Comment by wombat** [comment]
@riywo I also use bash `4.4.12(1)-release` and both versions work for me

**Comment by riywo** [comment]
Hmm. My locale is like below in my bash:

```bash
$ locale
LANG="en_US.UTF-8"
LC_COLLATE="en_US.UTF-8"
LC_CTYPE="en_US.UTF-8"
LC_MESSAGES="en_US.UTF-8"
LC_MONETARY="en_US.UTF-8"
LC_NUMERIC="en_US.UTF-8"
LC_TIME="en_US.UTF-8"
LC_ALL="en_US.UTF-8"
```

**Comment by arun-gupta** [comment]
Fixed by #200
